### ISSUE TITLE
Unable to compile and run

### ISSUE THREAD

**Issue by amerkoleci** [Roderik11/Spark#1]
Hi,
I saw you released the source code, the engine doesn't compile and doesn't even run, lot of assets and stuff is missing.

I was trying to take look at the UI stuff, would you be able to make it successfully run?

Thanks,
Amer

**Comment by Roderik11** [comment]
Oh hey Amer! Thanks for checking out my stuff.
It doesn't compile? That's really unfortunate.
Could you show me some or all of the errors you get that prevent compilation?

Greetings, Chris

**Comment by amerkoleci** [comment]
Hi Chris,
First its missing the Squid project, second it has reference to file paths to your PC probably :)

**Comment by Roderik11** [comment]
If you want to check out the UI we should focus on the Editor. Lets get Spark.dll and Spark.Editor compiled.
Here are the References needed for Spark and Spark.Editor:
![image](https://github.com/user-attachments/assets/63b99f3d-17d5-4aad-bf6b-084fef2f8aac)
![image](https://github.com/user-attachments/assets/fced2b43-bb69-42a7-8f85-0a78d153b1cc)

When both compiled, create a folder somewhere.
Change editors Program.cs line 48 to point to that folder.
In that folder create another folder called Resources.
Copy everthing from [here](https://github.com/Roderik11/Squid/tree/master/StandardSkinResources) to YourFolder/Resources.
In Spark.Editor/EditorTestScene.cs, remove everything except the editor camera (line 249).
Now you should be able to start the editor.

Let me know how that goes for you.

**Comment by Roderik11** [comment]
Oh yes you'll have to grab the Squid project seperately and add it your solution.
Excuse the hardcoded paths - there should be a project selection dialog instead (i will add one soon),

**Comment by amerkoleci** [comment]
Wouldn't be possible for you to have the repo running without hacks?

Otherwise no problem :)

Thanks!

**Comment by Roderik11** [comment]
I did not expect anyone to grab this and try to run it.
My intention was mostly to provide sourcecode for people to explore.
But i understand the screenshots give the impression one can just download, run and use it.
I will use some time over the holidays to clean up a bit.

**Comment by amerkoleci** [comment]
I was just curious about the UI capabilities, from screenshots its looks neat :)

**Comment by Roderik11** [comment]
Thanks, yes the UI is actually pretty neat (imho) and worth checking out.
I should probably provide a sample project for Squid alone that shows some stuff and is easy to compile&run.

**Comment by Roderik11** [comment]
And i've been a fan of your repos for a loooong time btw. It's cool that we're even talking.

**Comment by Roderik11** [comment]
Once the editor is able to start, you should be able to throw some assets into the Resources folder (.fbx, .obj, .x, .daer, all kinds of texture formats). Restart the editor, it will convert the assets to engine formats on startup (and store in Library folder).
Once you have assets show up in the editor you can inspect them and see some UI stuff in action.
Or in EditorTestScene, create some entites, so you have some stuff in the UI to select and interact with.
Or you can throw in a 16bit heightmap (should be pow2+1) and render some terrain.
There are plenty examples of entities in EditorTestScene, Client/GameTestScene and Spark/SceneUtil.cs

**Comment by amerkoleci** [comment]
> And i've been a fan of your repos for a loooong time btw. It's cool that we're even talking.

Glad to hear that, you could reach out and talk to me whether you want, even on discord if you want 

Cheers!

**Comment by Roderik11** [comment]
Oh one more thing, you need texconv.exe (https://github.com/microsoft/DirectXTex) in the same folder as the editor executable.
I'm using to convert textures, but it should really do it internally and not via command line.
Sitll working on that :D

**Comment by amerkoleci** [comment]
Ok,
I saw you have a roadmap on doing bindless and other refactoring, are those in plans?

**Comment by Roderik11** [comment]
Yeah absolutely. I've been undecided which way to go for a long time, Vulkan, DX12 or WebGPU.
I think i'm finally ready to go with DX12 and most likely your bindinds (which are closest to SharpDX so it feels familiar).
And i've been sinking way too much time into GPU frustum and occlusion culling with a HZ buffer.
So yeah it's def in the works, but we all only have 2 arms and too few hours in the day ;)

**Comment by Roderik11** [comment]
You can find me on Discord as Rod#1445
